src/main/java/de/hs_mannheim/informatik/ct/controller/RoomController.java

```diff
@@ -156,16 +156,34 @@ public String checkoutPage(@PathVariable String roomId, Model model) {
         if (!room.isPresent()) {
             throw new RoomNotFoundException();
         }
-
         Room.Data roomData = new Room.Data(room.get());
+        model.addAttribute("room", room.get());
+        model.addAttribute("checkout", true);
+        model.addAttribute("roomVisitService", roomVisitService);
         model.addAttribute("roomData", roomData);
         model.addAttribute("visitData", new RoomVisit.Data(roomData));
-
-        // The check-in page can handle both check-in and checkout with a css toggle
-        model.addAttribute("checkout", true);
+        model.addAttribute("privileged", false);
         return "rooms/checkIn";
     }
 
+    @GetMapping("/{roomId}/roomReset")
+    public String roomReset(@PathVariable String roomId, Model model) {
+        Optional<Room> room = roomService.findByName(roomId);
+        if (!room.isPresent()) {
+            throw new RoomNotFoundException();
+        }
+        Room.Data roomData = new Room.Data(room.get());
+        model.addAttribute("roomData", roomData);
+        return "rooms/roomReset";
+    }
+
+    @GetMapping("/{roomId}/executeRoomReset")
```

Comment:
HTTP Requests die etwas 'tun', also den State der Anwendung verändern respektive in die DB schreiben, müssen immer `POST` (Bzw. in APIs auch `PUT`, `DELETE`...) Requests sein. Das hat zum einen Sicherheitshintergründe -> [CSRF](https://owasp.org/www-community/attacks/csrf) und zu verhindern das die Request von einem Cache bedient wird.